pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -178,21 +181,60 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 	return utilerrors.NewAggregate(errs)
 }
 
+// validateCertificateSAN makes sure that a given certificate is valid
+// at least for the SANs defined in the configuration.
+func (c *EtcdCertSignerController) validateCertificateSAN(certName string, ipAddresses []string) error {
+	secret, err := c.secretLister.Secrets(operatorclient.TargetNamespace).Get(certName)
+	if err != nil {
+		return fmt.Errorf("could not get secrets from %s: %w", certName, err)
+	}
+	if secret.Data == nil {
+		return fmt.Errorf("secret data not found in %s", certName)
+	}
+	tlsCrt, ok := secret.Data["tls.crt"]
+	if !ok {
+		return fmt.Errorf("tls cert data not found in %s", certName)
+	}
+	if len(tlsCrt) == 0 {
+		return fmt.Errorf("%s missing tls.crt content\n", certName)
+	}
+	certificates, err := cert.ParseCertsPEM([]byte(tlsCrt))
```

Comment:
I think you could simplify this a bit by just passing `secret.Data` directly to `cert.ParseCertsPEM`. The function handles the error checks you manually added above by looking at the code[1].

[1]https://github.com/kubernetes/client-go/blob/master/util/cert/pem.go#L35